src/app/progress/progress/progress.component.ts

```diff
@@ -171,7 +178,21 @@ export class ProgressComponent implements OnInit {
     }
   }
 
-  averageStudentMark(servData) {
+  getAverageStudentMark(servData) {
+    let avrStudentMark = 0;
+
+    // find average student mark by subject
+    if (servData.length !== 0) {
+      servData.forEach((value: any) => {
+        if (value.subjectId === this.selectedSubjects.subjectId) {
+          avrStudentMark = value.avgMark;
+        }
+      });
+    }
+    return avrStudentMark;
```

Comment:
return servData.find(item => item.subjectId === this.selectedSubjects.subjectId).avgMark || 0;

Reply:
changed code